asset-transfer-secured-agreement/application-gateway-typescript/src/contractWrapper.ts

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Contract } from '@hyperledger/fabric-gateway';
+import { TextDecoder } from 'util';
+import { Asset, AssetJSON, AssetPrice, AssetPriceJSON, AssetPrivateData, AssetProperties, AssetPropertiesJSON, GREEN, parse, RED, RESET } from './utils';
+
+export class ContractWrapper {
+
+    readonly #contract: Contract;
+    readonly #org: string;
+    readonly #utf8Decoder  = new TextDecoder();
+    readonly #randomBytes: string;
+
+    public constructor(contract: Contract, org: string, randomBytes: string) {
+        this.#contract = contract;
+        this.#org = org;
+        this.#randomBytes = randomBytes;
+    }
+
+    public async createAsset(asset: Asset, privateData: AssetPrivateData): Promise<void> {
+        console.log(`${GREEN}--> Submit Transaction: CreateAsset, ${asset.AssetId} as ${asset.OwnerOrg} - endorsed by Org1${RESET}`);
+        try {
+            const assetPropertiesJSON: AssetPropertiesJSON = Object.assign({}, {objectType: 'asset_properties',
+                assetID: asset.AssetId,
+                color: privateData.Color,
+                size: privateData.Size,
+                salt: this.#randomBytes });
```

Comment:
Object.assign() is only really useful here if you have an object that is a subset of (or at least has only property names that match) the object you want to produce, and that you want to augment with additional properties (or modify existing properties). Or as a shallow copy of an object. Here it's not doing anything useful.
```suggestion
            const assetPropertiesJSON: AssetPropertiesJSON = {
                objectType: 'asset_properties',
                assetID: asset.AssetId,
                color: privateData.Color,
                size: privateData.Size,
                salt: this.#randomBytes,
            });
```